### PR TITLE
All x64_86 CPU should support SSE2

### PR DESCRIPTION
**Pull request by SergiySW**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,9 +34,11 @@ else (WIN32)
     endif()
 
 	IF (CMAKE_SYSTEM_PROCESSOR MATCHES "^(i.86|x86(_64)?)$")
-        if (RAIBLOCKS_SIMD_OPTIMIZATIONS)
-		    set (PLATFORM_COMPILE_FLAGS "${PLATFORM_COMPILE_FLAGS} -msse4")
-        endif()
+		if (RAIBLOCKS_SIMD_OPTIMIZATIONS)
```
Comment (androm3da) [inline CMakeLists.txt]: This conditional `MATCHES "^(i.86|x86(_64)?)$"` includes non-`x86_64` targets.

**Comment by SergiySW** [comment]
In fact 32-bit CPU aren't supported anymore

**Comment by lukealonso** [comment]
Ideally we would check for compiler support for SSE2/4 and AVX2 in CMakeLists (not hard), and refactor blake2b and ed25519 to compile all of the various supported backends instead of just one, then do a runtime check a la:

https://github.com/twitter/vireo/blob/master/imagecore/image/internal/platform_support.cpp

File: CMakeLists.txt [hunk 2]
```diff
@@ -34,9 +34,11 @@ else (WIN32)
...
+			set (PLATFORM_COMPILE_FLAGS "${PLATFORM_COMPILE_FLAGS} -msse4")
+		elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
+			set (PLATFORM_COMPILE_FLAGS "${PLATFORM_COMPILE_FLAGS} -msse2")
```
Comment (clemahieu) [inline CMakeLists.txt]: Do we need to set -msse2 here if it's set already above?

Comment (SergiySW) [reply CMakeLists.txt]: Where?

File: CMakeLists.txt [hunk 3]
```diff
@@ -21,7 +21,7 @@ if(NOT CMAKE_BUILD_TYPE)
 endif(NOT CMAKE_BUILD_TYPE)
 
 if (WIN32)
-	set (PLATFORM_COMPILE_FLAGS "-DBOOST_SPIRIT_THREADSAFE -D_WIN32_WINNT=0x0600 -DWINVER=0x0600 -DWIN32_LEAN_AND_MEAN -DMINIUPNP_STATICLIB")
+	set (PLATFORM_COMPILE_FLAGS "-DBOOST_SPIRIT_THREADSAFE -D_WIN32_WINNT=0x0600 -DWINVER=0x0600 -DWIN32_LEAN_AND_MEAN -DMINIUPNP_STATICLIB -msse2")
```
Comment (clemahieu) [inline CMakeLists.txt]: -msse2 here

Comment (SergiySW) [reply CMakeLists.txt]: For Windows

**Pull request merged**